doc/storage/api/api.rst

```diff
@@ -313,6 +339,139 @@ These definitions must be defined in the header file :file:`psa/internal_trusted
+       
+   This function must be fully defined if iteration is supported.
+       
+   If iteration is not supported, then this function SHALL be defined in the header - potentially as an inline function - but SHALL return ``PSA_ERROR_NOT_SUPPORTED``
+
+   .. param::  psa_its_storage_iterator_t *context
+       The location in which the function should store a pointer to the newly created context for this iterator. 
+       This is set to a new value on success and is undefined on error. 
+       The content of the iterator is implementation defined.
+
+   .. param:: psa_storage_uid_t filter
+       A value used to filter the results included in this iteration.
+
+   .. param:: int_t filter_length
+       A length of the filter to use, this must be a value ``0 < filter_length < 63``.
+
+   .. param:: psa_storage_uid_t *result
+        A pointer to the location in which to store ``uid``. On success the contents of this location will be updated with the first matching ``uid``. On error, the contents are undefined.
+
+   .. return:: psa_status_t
+       A status indicating the success or failure of the operation.
+
+   .. retval:: PSA_SUCCESS
+       The operation completed successfully.
+
+   .. retval:: PSA_ERROR_DOES_NOT_EXIST
+       No ``uid`` matches this iteration.
+
+    .. retval:: PSA_ERROR_INSUFFICIENT_STORAGE
+        The operation failed because there was insufficient space on the storage medium to create a new context.
+
+   .. retval:: PSA_ERROR_STORAGE_FAILURE
+       The operation failed because the physical storage has failed (Fatal error).
+
+
+.. function:: psa_its_iterator_next
+
+   .. summary::
+
+      Returns the next ``uid`` in this iteration.
+      This function must be fully defined if iteration is supported.
+      If iteration is not supported, then this function SHALL be defined in the header - potentially as an inline function - but SHALL return ``PSA_ERROR_NOT_SUPPORTED``
+
+
+   .. param::  psa_its_storage_iterator_t* context
+       A pointer to a context for this iterator as returned by `psa_its_iterator_start()` or updated by a previous call to `psa_its_iterator_next()`.  The content of the iterator will change on success and is undefined on error.
+
+   .. param:: psa_storage_uid_t *result
+        A pointer to the location in which to store ``uid``. On success the contents of this location will be updated with the next matching ``uid``. On error, the contents are undefined.
+
+   .. return:: psa_status_t
+       A status indicating the success or failure of the operation.
+
+   .. retval:: PSA_SUCCESS
+       The operation completed successfully.
+
+   .. retval:: PSA_ERROR_DOES_NOT_EXIST
+       The iterator has returned all the uids that match this iteration.
+
+   .. retval:: PSA_ERROR_STORAGE_FAILURE
+       The operation failed because the physical storage has failed (Fatal error).
+
+   .. retval:: PSA_ERROR_DATA_CORRUPT
```

Comment:
This contradicts the description text relating to the stability of the iterator when the underlying data storage is modified by the application between calls?